qiskit_nature/problems/second_quantization/lattice/lattice/lattice.py

```diff
@@ -0,0 +1,354 @@
+
+    @classmethod
+    def from_nodes_edges(
+        cls, num_nodes: int, weighted_edges: List[Tuple[int, int, complex]]
+    ) -> "Lattice":
+        """Return an instance of Lattice from the number of nodes and the list of edges.
+
+        Args:
+            num_nodes: The number of nodes.
+            weighted_edges: A list of tuples consisting of two nodes and the weight between them.
+        Returns:
+            Lattice generated from lists of nodes and edges.
+        """
+        graph = PyGraph(multigraph=False)
+        graph.add_nodes_from(range(num_nodes))
+        graph.add_edges_from(weighted_edges)
+        return cls(graph)
+
+    def to_adjacency_matrix(self, weighted: bool = True) -> np.ndarray:
+        """Return the hopping matrix from weighted edges.
+        The weighted edge list is interpreted as the upper triangular matrix.
+        Args:
+            weighted: The matrix elements are 0 or 1 when it is true.
+                Otherwise, the weights on edges are returned as a matrix.
+        Returns:
+            The adjacency matrix of the input graph.
+        """
+        if weighted:
+            real_part = adjacency_matrix(self.graph, weight_fn=np.real)
+            imag_part = adjacency_matrix(self.graph, weight_fn=np.imag)
+            imag_part = np.triu(imag_part) - np.triu(imag_part).T
+            ad_mat = real_part + 1.0j * imag_part
+        if not weighted:
+            ad_mat = adjacency_matrix(self.graph, weight_fn=lambda x: 1)
+
+        return ad_mat
+
+    def draw(
+        self,
+        pos: Optional[dict] = None,
+        ax: Optional[Axes] = None,
+        arrows: bool = True,
+        arrowstyle: Optional[str] = None,
+        arrow_size: int = 10,
+        with_labels: bool = False,
+        node_list: Optional[list] = None,
+        edge_list: Optional[list] = None,
+        node_size: Union[int, list] = 300,
+        node_color: Union[
+            str,
+            Tuple[float, float, float],
+            Tuple[float, float, float],
+            List[Union[str, Tuple[float, float, float], Tuple[float, float, float, float]]],
+        ] = "#1f78b4",
+        node_shape: str = "o",
+        alpha: Optional[float] = None,
+        cmap: Optional[Colormap] = None,
+        vmin: Optional[float] = None,
+        vmax: Optional[float] = None,
+        linewidths: Union[float, Sequence] = 1.0,
+        width: Union[float, Sequence] = 1.0,
+        edge_color: Union[str, Sequence] = "k",
+        edge_cmap: Optional[Colormap] = None,
+        edge_vmin: Optional[float] = None,
+        edge_vmax: Optional[float] = None,
+        style: str = "solid",
+        labels: Optional[Callable] = None,
+        edge_labels: Optional[Callable] = None,
+        font_size: int = 12,
+        font_color: str = "k",
+        font_weight: str = "normal",
+        font_family: str = "sans-serif",
+        label: Optional[str] = None,
+        connectionstyle: str = "arc3",
+        self_loop: bool = False,
```

Comment:
This subclass changes the function definition because it does *not* include `boundary_edges` as the last named argument. Although there is `**kwargs` there must be a better way to handle this.